firedrake/slate/preconditioners.py

```diff
@@ -124,6 +124,12 @@ def initialize(self, pc):
             trace_bcs = [DirichletBC(TraceSpace, Constant(0.0), "on_boundary")]
             K = Tensor(gammar('+') * ufl.dot(sigma, n) * ufl.dS)
 
+        # If boundary conditions are contained in the ImplicitMatrixContext:
+        if self.cxt.row_bcs:
+            raise NotImplementedError(
+                "Strong BCs not currently handled. Try imposing them weakly."
+            )
+
```

Comment:
Strong boundary conditions coming through is not a straightforward: "apply bcs to function then project into the broken space." I have tried many approaches and I simply cannot get this right.

The dofs associated with a non-broken HDiv space are topologically associated with the boundary facets, while the broken HDiv space only has "cell" dofs in this case. There doesn't seem to be a straightforward mapping yet which allows one to safely bring the bcs into the broken space. I have tried the naive "project them in" approach and it still doesn't get it right.

For now, I raise this exception until I figure out what needs to be done. Any help would be greatly appreciated.